pkg/frontend/transport/handler.go

```diff
@@ -252,6 +264,80 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getQueryDetail(url string, values url.Values) QueryDetail {
+	urlValues := parseURLValues(values)
+	return QueryDetail{
+		QueryString: urlValues.query,
+		StepInt:     urlValues.stepInt,
+		StartInt:    urlValues.startInt,
+		EndInt:      urlValues.endInt,
+		TsInt:       urlValues.tsInt,
+	}
+}
+
+type urlValues struct {
+	query                            string
+	stepInt, startInt, endInt, tsInt int64
+}
+
+func parseURLValues(values url.Values) urlValues {
+	query := values.Get("query")
+	step := values.Get("step")
+	start := values.Get("start")
+	end := values.Get("end")
+	ts := values.Get("time")
+	var (
+		tsInt, startInt, endInt, stepInt int64
+		err                              error
+	)
+	if len(step) == 0 { // instance query
+		if len(ts) > 0 {
+			tsInt, err = util.ParseTime(ts)
+			if err != nil {
+				util_log.Logger.Log("msg", "failed to parse time", "err", err)
```

Comment:
Can use leveled logging and `util_log.WithContext(r.Context(), f.log)`? This will also log the request ID if set. Like https://github.com/cortexproject/cortex/blob/master/pkg/frontend/transport/handler.go#L255